source/containers.tex

```diff
@@ -21425,3 +21490,631 @@
 swap(x.@\exposid{acc_}@, y.@\exposid{acc_}@);
 \end{codeblock}
 \end{itemdescr}
+
+\rSec3[mdspan.submdspan]{\tcode{submdspan}}
+
+\rSec4[mdspan.submdspan.overview]{Overview}
+
+\pnum
+The \tcode{submdspan} facilities create a new \tcode{mdspan}
+viewing a subset of elements of an existing input \tcode{mdspan}.
+The subset viewed by the created \tcode{mdspan} is determined by
+the \tcode{SliceSpecifier} arguments.
+
+\pnum
+For each function defined in subclause \ref{mdspan.submdspan} that
+takes a parameter pack named \tcode{slices} as an argument:
+
+\begin{itemize}
+\item let \tcode{index_type} be
+  \begin{itemize}
+  \item
+  \tcode{M::index_type}
+  if the function is a member of a class \tcode{M},
+  \item
+  otherwise,
+  \tcode{remove_reference_t<decltype(src)>::index_type}
+  if the function has a parameter named \tcode{src},
+  \item
+  otherwise,
+  the same type as the function's template argument \tcode{IndexType};
+  \end{itemize}
+\item let \tcode{rank} be the number of elements in \tcode{slices};
+\item let $s_k$ be the $k^\text{th}$ element of \tcode{slices};
+\item let $S_k$ be the type of $s_k$; and
+\item let \tcode{\placeholder{map-rank}} be an \tcode{array<size_t, rank>} such that
+for each $k$ in the range \range{0}{rank},
+\tcode{\placeholder{map-rank}[$k$]} equals:
+  \begin{itemize}
+  \item
+  \tcode{dynamic_extent}
+  if $S_k$ models \tcode{\libconcept{convertible_to}<index_type>},
+  \item
+  otherwise,
+  the number of types $S_j$ with $j < k$ that
+  do not model \tcode{\libconcept{convertible_to}<index_type>}.
+  \end{itemize}
+\end{itemize}
+
+\rSec4[mdspan.submdspan.strided.slice]{\tcode{strided_slice}}
+
+\pnum
+\tcode{strided_slice} represents a set of
+\tcode{extent} regularly spaced integer indices.
+The indices start at \tcode{offset}, and
+increase by increments of \tcode{stride}.
+
+\indexlibraryglobal{strided_slice}%
+\begin{codeblock}
+namespace std {
+  template<class OffsetType, class ExtentType, class StrideType>
+  struct strided_slice {
+    using offset_type = OffsetType;
+    using extent_type = ExtentType;
+    using stride_type = StrideType;
+
+    [[no_unique_address]] offset_type offset{};
+    [[no_unique_address]] extent_type extent{};
+    [[no_unique_address]] stride_type stride{};
+  };
+}
+\end{codeblock}
+
+\pnum
+\tcode{strided_slice} has the data members and special members specified above.
+It has no base classes or members other than those specified.
+
+\pnum
+\mandates
```

Comment:
Hm, now that we got rid of the itemdecl/itemdescr, it's odd to have a `\mandates` here.

Reply:
@jensmaurer Can we say "OffsetType, ExtentType, and StrideType shall be signed or unsigned integer types, or model integral-constant-like; otherwise the program is ill-formed."?

Reply:
Filed #6397.